pkg/cri/resource-manager/cache/cache.go

```diff
@@ -645,6 +653,120 @@ func (cch *cache) GetConfig() *config.RawConfig {
 	return cch.Cfg
 }
 
+// SetAdjustment updates external adjustments and containers based on this.
+func (cch *cache) SetAdjustment(external *config.Adjustment) (bool, map[string]error) {
+	effective := map[*container]string{}
+
+	// collect per container external adjustments, checking for obvious errors
+	errors := map[string]error{}
+	for id, c := range cch.Containers {
+		if id != c.GetCacheID() {
+			continue
+		}
+
+		adjustments := cch.getApplicableAdjustments(external, c)
+
+		if len(adjustments) == 0 {
+			continue
+		}
+
+		// conflict: multiple adjustments per container
+		if len(adjustments) > 1 {
+			errors[c.GetID()] = cacheError("conflicting adjustments for %s: %s",
+				c.PrettyName(), strings.Join(adjustments, ","))
+			continue
+		}
+
+		adjust := external.Adjustments[adjustments[0]]
+
+		// error: trying to override resources for BestEffort container
+		if c.GetQOSClass() == v1.PodQOSBestEffort {
+			if adjust.Resources != nil {
```

Comment:
No. This is singled out because (I tested it and) it wouldn't work. Some entity (probably ContainerManager in kubelet, but I haven't looked into it) sets and actively resets periodically `cpu.shares` in the entire besteffort cgroup subtree to `2`.

Reply:
Ok. Let's make a note to check if there are any other surprises waiting...